Framework/DataHandling/src/LoadNexusProcessed.cpp

```diff
@@ -577,33 +572,29 @@ void LoadNexusProcessed::correctForWorkspaceNameClash(std::string &wsName) {
 }
 
 /**
-* Check if the workspace name contains a common stem and load the workspace
-*names
+* Extract the workspace names from the file (if any are stored)
 *
 * @param root :: the root for the NeXus document
 * @param names :: vector to store the names to be loaded.
-* @return Whether there was a common stem.
 */
-bool LoadNexusProcessed::checkForCommonNameStem(
+void LoadNexusProcessed::extractWorkspaceNames(
     NXRoot &root, std::vector<std::string> &names) {
-  bool success(true);
   size_t nWorkspaceEntries = root.groups().size();
+  std::ostringstream os;
   for (size_t p = 1; p <= nWorkspaceEntries; ++p) {
-    std::ostringstream os;
+    os.str(std::string());
```

Comment:
I think `boost::lexical_cast` might make more sense here and might be a bit cleaner than using a stream. I also imagine there are optimizations for single integer values like this.